src/Application.vala

```diff
@@ -48,6 +48,62 @@ public class SettingsDaemon.Application : GLib.Application {
         add_main_option_entries (OPTIONS);
 
         housekeeping = new Backends.Housekeeping ();
+
+        var check_firmware_updates_action = new SimpleAction ("check-firmware-updates", null);
+        check_firmware_updates_action.activate.connect (() => {
+            var fwupd_client = new Fwupd.Client ();
+            var num_updates = 0;
+            try {
+                var devices = fwupd_client.get_devices ();
+                for (int i = 0; i < devices.length; i++) {
+                    var device = devices[i];
+                    if (device.has_flag (Fwupd.DEVICE_FLAG_UPDATABLE)) {
+                        Fwupd.Release? release = null;
+                        try {
+                            var upgrades = fwupd_client.get_upgrades (device.get_id ());
+
+                            if (upgrades != null) {
+                                release = upgrades[0];
+                            }
+                        } catch (Error e) {
+                            warning (e.message);
+                        }
+
+                        if (release != null && device.get_version () != release.get_version ()) {
```

Comment:
You probably could fold this into the if statement a couple lines up and not even need `release` anymore. Very optional change, I'll leave it up to you.